From 5a89f6f87720abfb14740beca4a1f5b0b48334ee Mon Sep 17 00:00:00 2001
From: Adam Ford <aford173@gmail.com>
Date: Tue, 22 Sep 2020 10:32:36 -0500
Subject: [PATCH 26/29] arm64: dts: beacon-renesom-som: Fix RZ/G2H and RZ/G2N
 Booting

A memory node for one of the memory banks on the common SOM dtsi
is not correct for booting the N and H variants.

This patch moves the node from the common SOM dtsi file into the proper
kit file for the RZ/G2M, and changes the address to permit booting
on the RZ/G2H and RZ/G2N.

Signed-off-by: Adam Ford <aford173@gmail.com>
---
 arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi       | 5 -----
 arch/arm64/boot/dts/renesas/r8a774a1-beacon-rzg2m-kit.dts | 5 +++++
 arch/arm64/boot/dts/renesas/r8a774e1-beacon-rzg2h-kit.dts | 5 +++++
 3 files changed, 10 insertions(+), 5 deletions(-)

diff --git a/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi b/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi
index 2f2ded41445d..6016d2e13445 100644
--- a/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi
+++ b/arch/arm64/boot/dts/renesas/beacon-renesom-som.dtsi
@@ -15,11 +15,6 @@
 		reg = <0x0 0x48000000 0x0 0x78000000>;
 	};
 
-	memory@600000000 {
-		device_type = "memory";
-		reg = <0x6 0x00000000 0x0 0x80000000>;
-	};
-
 	reserved-memory {
 		#address-cells = <2>;
 		#size-cells = <2>;
diff --git a/arch/arm64/boot/dts/renesas/r8a774a1-beacon-rzg2m-kit.dts b/arch/arm64/boot/dts/renesas/r8a774a1-beacon-rzg2m-kit.dts
index 966f0e2c3aa8..86b45d078083 100644
--- a/arch/arm64/boot/dts/renesas/r8a774a1-beacon-rzg2m-kit.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774a1-beacon-rzg2m-kit.dts
@@ -22,4 +22,9 @@
 	chosen {
 		stdout-path = "serial0:115200n8";
 	};
+
+	memory@600000000 {
+		device_type = "memory";
+		reg = <0x6 0x00000000 0x0 0x80000000>;
+	};
 };
diff --git a/arch/arm64/boot/dts/renesas/r8a774e1-beacon-rzg2h-kit.dts b/arch/arm64/boot/dts/renesas/r8a774e1-beacon-rzg2h-kit.dts
index 9131bb933025..1c3463ce7320 100644
--- a/arch/arm64/boot/dts/renesas/r8a774e1-beacon-rzg2h-kit.dts
+++ b/arch/arm64/boot/dts/renesas/r8a774e1-beacon-rzg2h-kit.dts
@@ -22,4 +22,9 @@
 	chosen {
 		stdout-path = "serial0:115200n8";
 	};
+
+	memory@500000000 {
+		device_type = "memory";
+		reg = <0x5 0x00000000 0x0 0x80000000>;
+	};
 };
-- 
2.17.1

